### PR TITLE
#434 add premium V3,

### LINKED ISSUE #434: Add PremiumV3 skus for web apps
PremiumV3 skus were added for app service plans were added recently. 

Adding support for these at basic level looks as though it may be as simple as adding another case to WebApp Sku in Common.fs

```fs
type Sku =
        | Shared
        | Free
        | Basic of string
        | Standard of string
        | Premium of string
        | PremiumV2 of string
        | PremiumV3 of string
        | Isolated of string
        | Dynamic
```

And then adding the appropriate members

```fs
 static member P1V3 = PremiumV3 "P1V3"
 static member P2V3 = PremiumV3 "P2V3"
 static member P3V3 = PremiumV3 "P3V3"
```

Still poking around and familiarizing myself with the code, so I may be missing something more that would need to be changed.

 I'd be happy to submit a pr for this change.

### PR DESCRIPTION
**Pull request by MNie**
- update cognitive services SKU

This PR closes #434

The changes in this PR are as follows:

* Add premiumV3 to WebApp
* Add S5/S6/S7/S8 for CognitiveServices

I have read the [contributing guidelines](CONTRIBUTING.md) and have completed the following:

* [ ] **Tested my code** end-to-end against a live Azure subscription.
* [X] **Updated the documentation** in the docs folder for the affected changes.
* [X] **Written unit tests** against the modified code that I have made.
* [X] **Updated the [release notes](RELEASE_NOTES.md)** with a new entry for this PR.
* [X] **Checked the coding standards** outlined in the [contributions guide](CONTRIBUTING.md) and ensured my code adheres to them.

If I haven't completed any of the tasks above, I include the reasons why here:

CC: @isaacabraham 

I saw that we don't give an ability to select S5/S6/S7/S8 for Bing.Search as noted [here](https://docs.microsoft.com/en-us/rest/api/cognitiveservices/accountmanagement/resourceskus/list). I add this option but I also include a validation regarding the kind & sku because S5/S6/S7/S8 are available only for Bing.Search.

I also update the plans for WebApp in the same PR.

Questions I have:
- Is it okay to do it in a single PR or should I create separate PRs?
- Validation for kind & sku seems reasonable or we don't want to do this?

Because of the above questions, I created the PR as a draft.

### CODE REVIEW

File: src/Tests/CognitiveServices.fs [hunk 1]
```diff
@@ -12,6 +12,41 @@ open System
 let dummyClient = new CognitiveServicesManagementClient (Uri "http://management.azure.com", TokenCredentials "NotNullOrWhiteSpace")
 let getResourceAtIndex o = o |> getResourceAtIndex dummyClient.SerializationSettings
 
+let casesForSkuLessThanPremium =
```
Comment (isaacabraham) [inline src/Tests/CognitiveServices.fs:15]: Formatting, start / end lists should be as follows:

```fsharp
let x = [
    ...
    ...
]

File: src/Farmer/Common.fs [hunk 2]
```diff
@@ -420,6 +430,27 @@ module CognitiveServices =
         | TextAnalytics
         | TextTranslation
 
+    type CognitiveServicesSku =
+        private | CognitiveServicesSku of Sku
+        static member Create kind sku =
+            [
+                yield
```
Comment (isaacabraham) [inline src/Farmer/Common.fs:437]: `yield` keyword is obsolete.

**Review by isaacabraham — CHANGES_REQUESTED** [review]

**Comment by isaacabraham** [comment]
Great questions. I think on reflection it would be better to do the Cognitive Services stuff as a separate PR please.

Happy to merge in the App Service stuff, as long as you can spin up an instance for the new sku to prove it works (I'm sure if you're quick it won't cost more than 2 or 3 cents ;-))

**Comment by MNie** [comment]
@isaacabraham I extracted Cognitive services changes as a separate PR: #440 should I create also an Issue for that?

**Comment by isaacabraham** [comment]
Any chance that you can do an end-to-end test of this please?

**Comment by isaacabraham** [comment]
@MNie happy to merge this in once you've done an end-to-end on this.

**Comment by daniellittledev** [comment]
Are we just waiting on the  end-to-end test for this PR?

**Comment by isaacabraham** [comment]
@daniellittledev basically, yes.

**Comment by isaacabraham** [comment]
Thanks for this!

**Pull request merged**